tidy3d/components/autograd/utils.py

```diff
@@ -2,14 +2,37 @@
 
 import typing
 
-from autograd.tracer import getval
+import autograd.numpy as anp
+import numpy as np
+from autograd.tracer import getval, isbox
+
+
+def _is_traced_objary(x: typing.Any) -> bool:
+    """Check whether x is a traced object array."""
+    return isinstance(x, np.ndarray) and x.size > 0 and isbox(x.flat[0])
+
+
+def is_traced(x: typing.Any) -> bool:
+    """Check whether x is traced (contains an autograd Box)"""
+    return isbox(x) or _is_traced_objary(x)
 
 
 def get_static(x: typing.Any) -> typing.Any:
     """Get the 'static' (untraced) version of some value."""
     return getval(x)
 
 
+def get_tracer(x: typing.Any) -> typing.Any:
+    """Get the 'boxed' (traced) version of some value."""
+    if not is_traced(x):
+        return None
+
+    if not _is_traced_objary(x):
```

Comment:
pretty minor but slighly cleaner might be
```py
if _is_traced_objary(x):
     return anp.array(x.tolist())

return x
```
?

Reply:
good point